static/app/views/projectInstall/issueAlertNotificationOptions.tsx

```diff
@@ -0,0 +1,150 @@
+    action.map(a =>
+      a === IssueAlertActionType.NOTIFY_EMAIL
+        ? MultipleCheckboxOptions.EMAIL
+        : MultipleCheckboxOptions.INTEGRATION
+    )
+  );
+
+  const messagingIntegrationsQuery = useApiQuery<OrganizationIntegration[]>(
+    [`/organizations/${organization.slug}/integrations/?integrationType=messaging`],
+    {staleTime: Infinity}
+  );
+
+  const refetchConfigs = () => messagingIntegrationsQuery.refetch();
+
+  const providersToIntegrations = useMemo(() => {
+    const map: {[key: string]: OrganizationIntegration[]} = {};
+    if (!messagingIntegrationsQuery.data) {
+      return {};
+    }
+    for (const i of messagingIntegrationsQuery.data) {
+      const providerSlug = i.provider.slug;
+      map[providerSlug] = map[providerSlug] ?? [];
+      map[providerSlug].push(i);
+    }
+    return map;
+  }, [messagingIntegrationsQuery.data]);
+
+  useEffect(() => {
+    const providerKeys = Object.keys(providersToIntegrations);
+    if (providerKeys.length > 0) {
+      const firstProvider = providerKeys[0];
+      setProvider(firstProvider);
+
+      const firstIntegration = providersToIntegrations[firstProvider][0];
+      setIntegration(firstIntegration);
+    }
+  }, [providersToIntegrations, setProvider, setIntegration]);
+
+  const shouldRenderSetupButton = useMemo(() => {
+    return messagingIntegrationsQuery.data?.every(i => i.status !== 'active');
+  }, [messagingIntegrationsQuery.data]);
+
+  const shouldRenderNotificationConfigs = useMemo(() => {
+    return selectedValues.some(v => v !== MultipleCheckboxOptions.EMAIL);
+  }, [selectedValues]);
+
+  const onChange = values => {
+    setSelectedValues(values);
+    setAction(
+      values.map(v =>
+        v === MultipleCheckboxOptions.INTEGRATION && provider
+          ? providerDetails[provider].action
+          : IssueAlertActionType.NOTIFY_EMAIL
+      )
+    );
+  };
+
+  if (messagingIntegrationsQuery.isLoading || messagingIntegrationsQuery.isError) {
+    return null;
+  }
+
+  return (
+    <div>
+      <MultipleCheckbox name="notification" value={selectedValues} onChange={onChange}>
+        <Wrapper>
+          <MultipleCheckbox.Item value={MultipleCheckboxOptions.EMAIL} disabled>
+            Notify via email
+          </MultipleCheckbox.Item>
+          {!shouldRenderSetupButton && provider && (
+            <div>
+              <MultipleCheckbox.Item value={MultipleCheckboxOptions.INTEGRATION}>
+                Notify via integration (Slack, Discord, MS Teams, etc.)
+              </MultipleCheckbox.Item>
+              {shouldRenderNotificationConfigs && (
+                <MessagingIntegrationAlertRule
+                  providersToIntegrations={providersToIntegrations}
+                />
+              )}
+            </div>
+          )}
+        </Wrapper>
+      </MultipleCheckbox>
+      {shouldRenderSetupButton && (
+        <SetupMessagingIntegrationButton
+          refetchConfigs={refetchConfigs}
```

Comment:
could pass refetch directly since the refetchConfigs function doesn't add anything
```suggestion
          refetchConfigs={messagingIntegrationsQuery.refetch}
```